src/components/Lesson.vue

```diff
@@ -258,11 +304,66 @@ export default {
         return new IPFS({repo: Math.random().toString()})
       }
     },
+    resetCode: function () {
+      // console.log('in resetCode')
+      this.code = this.$attrs.code || defaultCode
+      // this ^ will trigger onCodeChange which will clear cache
+      this.editor.setValue(this.code)
+      this.clearPassed()
+    },
+    clearPassed: function () {
+      // console.log('in clearPassed')
+      // console.log('this.lessonPassed is: ', this.lessonPassed)
+      delete localStorage[this.lessonKey]
+      this.lessonPassed = !!localStorage[this.lessonKey]
+      // console.log('now lessonPassed is: ', this.lessonPassed)
+    },
+    loadCodeFromCache: function() {
+      this.code = localStorage[this.cacheKey]
+      this.editor.setValue(this.code)
+    },
     onMounted: function (editor) {
+      // runs on page load, NOT on every keystroke in editor
+      // console.log('in MOUNTED and localStorage[this.lessonKey] is: ', localStorage[this.lessonKey])
+      // console.log('this.lessonPassed is: ', this.lessonPassed)
+      // console.log('in MOUNTED and localStorage[this.cacheKey] is: ', localStorage[this.cacheKey])
+      // console.log('this.cachedCode is: ', this.cachedCode)
+      // don't need to update lessonPassed here because it sets itself correctly
+      // in 'data' on page load
       this.editor = editor
+      if (this.cachedCode) {
+        console.log('PRESUMPTION: Returned to a lesson previously cached (1)')
```

Comment:
From this point on down there are a lot of debugging statements still int he code.